src/coreComponents/physicsSolvers/surfaceGeneration/EmbeddedSurfaceGenerator.cpp

```diff
@@ -158,6 +158,16 @@ void EmbeddedSurfaceGenerator::InitializePostSubGroups( Group * const problemMan
 
   embeddedSurfaceSubRegion->inheritGhostRank( cellElemGhostRank );
 
+  dataRepository::Wrapper< SortedArray< localIndex > > * setWrapper =
+      embeddedSurfaceSubRegion->GetGroup( ObjectManagerBase::groupKeyStruct::setsString )->
+      getWrapper< SortedArray< localIndex > >( "all" );
+
+  SortedArray< localIndex > & targetSet = setWrapper->reference();
+  for (localIndex i=0; i < embeddedSurfaceSubRegion->size(); ++i)
+  {
+    targetSet.insert( i );
+  }
+
```

Comment:
@rrsettgast I have solved like this the issue with the set. If I don't do this the set is empty and I cannot set initial conditions for fields on the `embeddedSurfaceSubRegion`. It's not pretty but it works.

Reply:
Can't you add append to the set when you create new elements?